### PR TITLE
Update Kb (knowledgebase) documentation

### PR DESCRIPTION
**Pull request by kingthorin**
I've had this on my todo list for a while and mentioned it on IRC a number of times. Finally had a chance to tackle it.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Review by psiinon — APPROVED** [review]

File: docs/Kb.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+This file documents the places the `Kb` (Knowledgebase) are used within ZAP and zap-extensions. (`org.parosproxy.paros.core.scanner.Kb`)
+
+Use of `Kb` is currently exposed through `AbstractPlugin` and `HostProcess` `getKb()` methods.
+
+## URI Level
+
+- `org.zaproxy.zap.extension.ascanrules.SqlInjectionScanRule` tracks
+    - sql/<rdbms_name>
+- `org.zaproxy.addon.commonlib.AbstractAppFilePlugin` scan rules track completed 'tests' to prevent duplication.
```
Comment (thc202) [inline docs/Kb.md:9]: I don't think this should be documented, it's not meant to be used by anyone/anything other than the rules themselves and it's a workaround.

Comment (kingthorin) [reply docs/Kb.md:9]: Okay, I’ll remove that bit.

**Comment by kingthorin** [comment]
Fixed

File: docs/Kb.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+This file documents the places the `Kb` (Knowledgebase) are used within ZAP and zap-extensions. (`org.parosproxy.paros.core.scanner.Kb`)
```
Comment (kingthorin) [inline docs/Kb.md:1]: This should say “is” used…will tweak in the morning

Comment (thc202) [reply docs/Kb.md:1]: Ping.

Comment (kingthorin) [reply docs/Kb.md:1]: Thanks will tackle it shortly ;)

**Comment by kingthorin** [comment]
Done

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**